cmd/cli/desktop/desktop.go

```diff
@@ -335,6 +343,31 @@ func (c *Client) fullModelID(id string) (string, error) {
 		if m.ID[7:19] == id || strings.TrimPrefix(m.ID, "sha256:") == id || m.ID == id {
 			return m.ID, nil
 		}
+		// Check if the ID matches any of the model's tags using exact match first
+		for _, tag := range m.Tags {
+			if tag == id {
+				return m.ID, nil
+			}
+		}
+		// If not found with exact match, try partial name matching
+		for _, tag := range m.Tags {
+			// Extract the model name without tag part (e.g., from "ai/smollm2:latest" get "ai/smollm2")
+			tagWithoutVersion := tag
+			if idx := strings.LastIndex(tag, ":"); idx != -1 {
+				tagWithoutVersion = tag[:idx]
+			}
+
+			// Get just the name part without organization (e.g., from "ai/smollm2" get "smollm2")
+			namePart := tagWithoutVersion
+			if idx := strings.LastIndex(tagWithoutVersion, "/"); idx != -1 {
+				namePart = tagWithoutVersion[idx+1:]
+			}
+
+			// Check if the ID matches the name part
+			if namePart == id {
+				return m.ID, nil
+			}
```

Comment:
I think you can use `name.ParseReference(tag)` to get same results:
 ```suggestion
// Try to parse both the stored tag and the input ID
			tagRef, err1 := name.ParseReference(tag)
			idRef, err2 := name.ParseReference(id)

			if err1 != nil || err2 != nil {
				// fallback to string comparison if parsing fails
				if tag == id {
					return m.ID, nil
				}
				continue
			}

			// Compare normalized names (without tag or digest)
			if tagRef.Context().Name() == idRef.Context().Name() {
				return m.ID, nil
			}
```